### ISSUE TITLE
transitionTo to same state with {reload: false, notify: false, location: true} causing re-resolving.

### ISSUE THREAD

**Issue by vamseekm** [angular-ui/ui-router#1674]
I am trying to change query parameters without reloading the state (I know I can use reloadOnSearch to do that and I am trying to avoid using that.). And each call to  $state.transitionTo is re-resolving all functions.

Here is the example code.
http://plnkr.co/edit/ycjz3C?p=preview

Is it possible to change query parameters, $state.params without re-resolving ?

**Comment by SebastianM** [comment]
+1

**Comment by skirjak** [comment]
+1

**Comment by rtrompier** [comment]
+1

**Comment by nateabele** [comment]
I'm not sure I follow? `reloadOnSearch` does what you want, yet you're trying to avoid it?

**Comment by chmanie** [comment]
I've run into the same issue. When I do a

`
transitionTo('myState', { myParam: foo }, { location: true, notify: false });
`

The location is changed, but the controller for the state is not re-instantiated (which is exactly what I want). Yet the resolves are fired (which makes no sense because I have no access to the newly resolved objects in my not-again-instanciated controller).

`reloadOnSearch` may fix the problem for query parameters (I did not test that) but not for basic route parameters `/:myParam`.

Basically the resolves should not fire if notify is false, as you have no possibility to access the new resolves if the controller is not re-instaciated.

**Comment by nateabele** [comment]
> Basically the resolves should not fire if notify is false

I'm not sure where people keep getting this idea. `notify` suppresses events. It does nothing else.

**Comment by chmanie** [comment]
> It does nothing else.

That may be true. But in some cases it appears to have the already described side-effect that the controller is not instantiated (as I described above). But maybe that is an edge-case that only happened to me (for another reason).

**Comment by btm1** [comment]
@nateabele I'm having the same issue and yes you would avoid reloadOnSearch: false because it causes you to not be able to type a query param in the url and hit return and have it change... so you gain something and lose something... I've figured a way around this by doing something like https://github.com/angular-ui/ui-router/issues/64#issuecomment-130865490 but this also has it's downsides as there are edge cases where there is a race condition during the set timeout where if you change states fast enough while this is going on that you get unwanted params passed to the next state.

**Comment by oste** [comment]
@nateabele I think the reason not to use reloadOnSearch:false is because you could have a route that has some params that update the view while others just maintain state until the page is refreshed or linked.

A route like this comes to mind ```url: "/posts?tags&dialog",``` Here when the tags parameter changes the view should be updated to reflect the tags search, but when the dialog parameter changes we are just showing a dialog to the user so the view should not refresh. If the user refreshed the page or sent the link to a friend the dialog should show up though. We can make that check using ```$stateParams```.

This seems like something that should be possible in the view level. Here is a good api for me. What do you think?
```
<button ui-sref=".({ dialog: true })" ui-sref-opts="{reload: false}">Show Dialog</button>
```

**Comment by nateabele** [comment]
@oste Yup, agreed. And that'll be totally viable in 1.0.

**Comment by oste** [comment]
@nateabele is there a roadmap or timeline for 1.0 release?

**Comment by nateabele** [comment]
https://github.com/angular-ui/ui-router/labels/1.0

**Comment by canvas103** [comment]
what about move search params to child state (only has url)  
.state("app.search",{
      url:"?filter_text"
    });

**Comment by b4dnewz** [comment]
no updates on this?

**Comment by Ronileco** [comment]
any updates? is this still a thing?

**Comment by blowsie** [comment]
@Ronileco 
dont think so, is angular still a thing?  🤣

**Comment by b4dnewz** [comment]
@blowsie exactly

**Comment by Ronileco** [comment]
Lolz, figured out myself, wasn't the reload:false.
BTW, just my 2 cents - Technology that works well shouldn't be replaced just for the sake of replacing.

Thanks for the help!

**Comment by nateabele** [comment]
> Technology that works well shouldn't be replaced just for the sake of replacing.

Yup, agreed. 😉

**Comment by deepeeess** [comment]
I ran into this issue while trying to upgrade to the `1.x` code train.  Downgrading back to `^0.3.1` fixed my problem.  Also, I had to blow away `node_nodules` before rebuilding prior to the downgrade.  Hope this helps someone.

**Comment by christopherthielen** [comment]
`reloadOnSearch` was replaced by `dynamic` params in 1.0

https://ui-router.github.io/ng1/docs/latest/interfaces/params.paramdeclaration.html#dynamic

```js
const state = {
  name: 'posts',
  url: "/posts?tags&dialog",
  params: { 
    dialog: { dynamic: true }
  }
}
```
